SECEdgar/filings/cik_validator.py

```diff
@@ -0,0 +1,108 @@
+import warnings
+
+from SECEdgar.network_client import NetworkClient
+from SECEdgar.utils.exceptions import CIKError, EDGARQueryError
+
+
+class CIKValidator(object):
+    """Validates company tickers and/or company names based on CIK availability.
+
+    Used internally by the CIK class. Not intended for outside use.
+
+    Args:
+        lookups (Union[str, list, tuple]): List of tickers and/or company names for
+            which to find CIKs.
+        **kwargs: Any keyword arguments needed to be passed to
+            _EDGARBase (see class for more details).
+
+    .. versionadded:: 0.1.5
+    """
+
+    def __init__(self, lookups, client=None, **kwargs):
+        if isinstance(lookups, str):
+            self._lookups = [lookups]
+        else:
+            try:
+                if not all(type(o) is str for o in lookups):
+                    raise TypeError("CIKs must be given as string or iterable.")
```

Comment:
I don't think you need to type two times the error message if you are doing `exception` handling`.

```
# using `as error` will remove writing same message again 
except TypeError as error:
               # raising again doesn't make sense
                raise TypeError("CIKs must be given as string or iterable.")
                print(error) or log it
```

Reply:
You're right. I will change this.